src/olympia/blocklist/utils.py

```diff
@@ -193,12 +193,55 @@ def split_regex_to_list(guid_re):
     return GUID_SPLIT.split(trimmed)
 
 
+def disable_addon_for_block(block, timestamp):
+    """Disable appropriate addon versions that are affected by the Block, and
+    the addon too if 0 - *."""
+    from .models import Block
```

Comment:
Preliminary review: Not a big fan of the duplication of code this causes. You have to copy/paste a bunch of things the `ReviewUnlisted` class did, and in a completely different app too. Also, this is missing the python logging that ReviewUnlisted` did.

I wonder if it wouldn't be better to instantiate the correct review object (can this happen for listed add-ons?) and add a method there that would do what we need - keeping the reviewer specific stuff in the reviewer object, re-using the existing code to clear flags, set status, log activity etc.

Reply:
I'm not sure there's a review action we can re-use directly but I'll look into it.  (my first iteration called the helper methods on ReviewBase directly but circular import error and having to make too many changes led me to create a specialised version here).

And yes, it works on listed versions the same as unlisted versions (no such thing as listed add-ons 😉) - the blocklist is channel agnostic.

Reply:
https://github.com/mozilla/addons-server/pull/15435/commits/3a6f29517ba1e6a1ae4dc20aea63f8d53b6976eb